src/core/core-varlink.c

```diff
@@ -84,15 +84,16 @@ static int build_managed_oom_json_array_element(Unit *u, const char *property, s
                 mode = managed_oom_mode_to_string(c->moom_swap);
         else if (streq(property, "ManagedOOMMemoryPressure")) {
                 mode = managed_oom_mode_to_string(c->moom_mem_pressure);
-                use_limit = true;
+                use_limit_and_duration = true;
         } else
                 return -EINVAL;
 
         return sd_json_buildo(ret_v,
                               SD_JSON_BUILD_PAIR("mode", SD_JSON_BUILD_STRING(mode)),
                               SD_JSON_BUILD_PAIR("path", SD_JSON_BUILD_STRING(crt->cgroup_path)),
                               SD_JSON_BUILD_PAIR("property", SD_JSON_BUILD_STRING(property)),
-                              SD_JSON_BUILD_PAIR_CONDITION(use_limit, "limit", SD_JSON_BUILD_UNSIGNED(c->moom_mem_pressure_limit)));
+                              SD_JSON_BUILD_PAIR_CONDITION(use_limit_and_duration, "limit", SD_JSON_BUILD_UNSIGNED(c->moom_mem_pressure_limit)),
+                              SD_JSON_BUILD_PAIR_CONDITION(use_limit_and_duration, "duration", SD_JSON_BUILD_UNSIGNED(c->moom_mem_pressure_duration_usec)));
```

Comment:
i think we should suppress the two fields if they weren't actually set.

i.e. the condition should probably be `use_limit_and_duration && c->oom_mem_pressure_limit != UINT64_MAX`

Reply:
Note moom_mem_pressure_limit default setting is 0 so we would check `c->oom_mem_pressure_limit_duration_usec != USEC_INFINITY` vs. `c->moom_mem_pressure_limit > 0`